deploy/examples/monitoring/rbac.yaml

```diff
@@ -92,6 +92,7 @@ rules:
       - list
       - create
       - update
+      - delete
```

Comment:
Actually we don't need this `delete`. The mgr doesn't need access to the delete the service monitor. The operator already has the `delete` privilege above with the other role on line 20. I believe the issue is that the operator doesn't always have this rbac.yaml created, unless the monitoring feature is enabled. So I'd suggest we just silently fail the delete (make it a debug level log), instead of failing the reconcile.